common/src/services/account.service.ts

```diff
@@ -0,0 +1,124 @@
+import { AccountService as AccountServiceAbstraction } from '../abstractions/account.service';
+import { StorageService } from '../abstractions/storage.service';
+import { AccountStorageKey } from '../enums/accountStorageKey';
+
+import { Account } from '../models/domain/account';
+import { SettingStorageOptions } from '../models/domain/settingStorageOptions';
+
+export class AccountService implements AccountServiceAbstraction {
+    private accounts: Account[] = [];
+    private activeUserId: string;
```

Comment:
Why not just use an `[index]` (`int`) for `accounts`, and reference that way. Active Account, which seems like it is going to be called a LOT will reduce the cycles necessary having to use `find()` when it can instead use `[index]` to reference the appropriate account.

OR, use the `userId` as the indexer and make `accounts` a dynamic object graph where the `userId` is the property name of a property of type `Account`? This completely eliminates the use of `find()` as a pattern which I see in many of the methods here, reducing CPU overhead and being a little more straightforward. How that works in TypeScript, 🤷🏻 , but that's the pattern that may work better and then you can keep the `activeUserId` property as-is.

Reply:
I really appreciate this feedback, another thing I had not even begun to notice. I changed this up to use a [`Record`](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeystype), removing all the `find()`s and indexing using `userId` instead.